### PR TITLE
04 ticket

### PR DESCRIPTION
**Pull request by StevenJune**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,36 @@
 import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+//import { useState, useEffect } from "react";
```
Comment (mallionaire) [inline src/App.js]: Please do not include commented out code

File: src/components/AllArticles.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  const [articles, setArticles] = useState([]);
+  const [isLoading,setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  console.log('in All Articles');
+  useEffect(() => {
+   axios 
+     .get("https://be-my-news-example.herokuapp.com/api/articles")
+     .then(function (response) {
+       // handle success
+       console.log(response.statusText,'kkkkerror obj')
```
Comment (mallionaire) [inline src/components/AllArticles.jsx]: remove console.logs

File: src/components/AllArticles.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        setArticles(response.data.articles);
+       }
+     });
+  }, []);
+  console.log('end of useEffect')
+  if (isLoading) return <p>Loading...</p>;
+  if (isError) return <p>Error happening........</p>
+
+  return (
+    <table>
```
Comment (mallionaire) [inline src/components/AllArticles.jsx]: Please use other html tags to display your information. You will not be able to use `<table> `

File: src/components/Topics.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+import ReadArticlesByTopic from './ReadArticlesByTopic.jsx';
```
Comment (mallionaire) [inline src/components/Topics.jsx:3]: This is an empty file and causing an error in the console

**Review by mallionaire — CHANGES_REQUESTED** [review]
Hi Steven, 
You will not be able to use `<table>` for this project. 
Please change the way articles, users and topics are displayed to the user, you could use `<h2>`, `<h3>` ,`<h4>` and `<p>`

You will need to remove all your console.logs.
`ReadArticlesByTopic.jsx` is empty and has no code.

**Comment by StevenJune** [comment]
Hi Haz,
    I cannot read the below suggestions, it just show (see below) , which mean I cannot use <table> <th><tr><td></table>, right?
,
Thanks
Steven


________________________________

**Review by Augs0 — CHANGES_REQUESTED** [review]
As discussed in our messages on Slack, please change from a table to another html tag to display your data. Table formats are outdated and also now used only for its proper use: displaying data

File: src/components/AllArticles.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  const [isError, setIsError] = useState(false);
+  useEffect(() => {
+   axios 
+     .get("https://be-my-news-example.herokuapp.com/api/articles")
+     .then(function (response) {
+       if (!response.statusText === 'OK') {
+        setIsError(true)
+        setIsLoading(false);
+       } else {
+        console.log(response.data.articles);
```
Comment (Rayhaan2k9) [inline src/components/AllArticles.jsx]: Please remove the console log

File: src/components/ReadArticlesByTopic.jsx [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+
+function ReadArticlesByTopic() {
+
+}
+export default ReadArticlesByTopic;
```
Comment (Rayhaan2k9) [inline src/components/ReadArticlesByTopic.jsx:7]: If this file is empty, there is no need to commit it

File: src/components/SingleArticle.jsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+function SingleArticle(prop) {
+  const [articles, setArticles] = useState([prop.current_id]);
+  const [isLoading,setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  useEffect(() => {
+   axios 
+     .get("https://be-my-news-example.herokuapp.com/api/articles/" + prop.current_id)
+     .then(function (response) {
+       console.log(response.statusText,'response obj from singlearticle')
```
Comment (Rayhaan2k9) [inline src/components/SingleArticle.jsx]: Please remove the console log

File: src/components/SingleArticle.jsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+       if (!response.statusText === 'OK') {
+        setIsError(true)
+        setIsLoading(false);
+       } else {
+        console.log(response.data.articles);
```
Comment (Rayhaan2k9) [inline src/components/SingleArticle.jsx]: Please remove the console log

File: src/components/Topics.jsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  useEffect(() => {
+   axios 
+     .get("https://be-my-news-example.herokuapp.com/api/topics")
+     .then(function (response) {
+       console.log(response.statusText,'topics response obj')
+       if (!response.statusText === 'OK') {
+        setIsError(true)
+        setIsLoading(false);
+       } else {
+        console.log(response.data.topics);
```
Comment (Rayhaan2k9) [inline src/components/Topics.jsx]: Please remove console log

File: src/components/Users.jsx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+//import { Route, Routes } from 'react-router-dom';
+
+function Users(prop) {
+  const [users, setUsers] = useState([prop.currentUser]);
+  const [isLoading,setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  console.log('in ListUser');
+//   const [articles, setArticles] = useState([]);
+//   const [isLoading,setIsLoading] = useState(true);
+//   const [isError, setIsError] = useState(false);
```
Comment (Rayhaan2k9) [inline src/components/Users.jsx]: Please remove these commented out code

File: src/components/Users.jsx [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+        setIsError(true)
+        setIsLoading(false);
+       } else {
+        console.log(response.data.users);
+        setIsLoading(false);
+        setUsers(response.data.users);
+       }
+     });
+  }, []);
+  console.log('end of useEffect')
```
Comment (Rayhaan2k9) [inline src/components/Users.jsx]: Please remove the console log

File: src/components/Users.jsx [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+
+export default Users;
+//<div id="myDIV">
+//  <div class="item1">1</div>
+//  <div class="item2">2</div>
+//  <div class="item3">3</div>
+//  <div class="item4">4</div>
+//  <div class="item5">500</div>
+//  <div class="item6">6</div>
+//</div>
```
Comment (Rayhaan2k9) [inline src/components/Users.jsx]: Please remove the commented out code

**Review by Rayhaan2k9 — CHANGES_REQUESTED** [review]
Looking better now. Just need to remove the console logs and commented out code :)

**Review by Augs0 — APPROVED** [review]
Hi Steven, I'm approving as we can now see all articles and they are not in a table :) well done! The layout you have does need some fixing to work on all screen sizes as the list of articles hangs off the screen on smaller devices, but this is something you can fix later

**Pull request merged**